activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -284,7 +284,7 @@ def rename_column(table_name, column_name, new_column_name) #:nodoc:
       end
 
       def add_reference(table_name, ref_name, **options) # :nodoc:
-        super(table_name, ref_name, type: :integer, **options)
```

Comment:
It is actually to make primary key and reference columns identical type.

https://github.com/rails/rails/blob/7813fc2c1fa9cc18c0a383ee1bc8ed47c613dbe8/activerecord/test/cases/base_test.rb#L145-L150

https://github.com/rails/rails/blob/5266b6a1c71b27d6a1958bf3d5999bdcf4167a03/activerecord/test/cases/migration/references_statements_test.rb#L25-L30

See also #27334.

Reply:
Yes, I have seen those, so, we already have test cases for identical type check. Thanks.